scripts/reference/run_reference_decode.py

```
@@ -0,0 +1,215 @@
+        sys.stdout.write(payload)
+        return
+    path.write_text(payload, encoding="utf-8")
+
+
+def require_list(name: str, value: Any) -> List[Any]:
+    if not isinstance(value, list):
+        raise ValueError(f"{name} must be a JSON array")
+    return value
+
+
+def require_number_list(name: str, value: Any) -> List[float]:
```

Ruff (0.15.10)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)